ndmg/stats/qa_tensor.py

```diff
@@ -33,86 +34,167 @@
+    
+    To show a slice, provide a sliced volume of the peak_dirs and peak_values and adjust the x_angle and y_angle 
+    See Tractography Directional Field QA Tutorial for examples
+    
+    Parameters
+    -----------
+    peak_dirs: np array of peak_dirs 
+    peak_values: np array of peak_values
+    slices: dictionary of lists of slices in x,y,z plane
+    x_angle: angle to rotate image along x axis
+    y_angle: angle to rotate image along y axis
+    fname: str path to save
+    size: size tuple for each image
+    """    
+    centers,directions,directions_colors,heights = generate_3_d_directions(peak_dirs,peak_values)
+    
+    scene = window.Scene()
+    arrow_actor = actor.arrow(centers, directions, directions_colors, heights)
+
+    scene.add(arrow_actor)
+    scene.roll(x_angle)
+    scene.pitch(y_angle)
+    
+    window.show(scene, size = size)
+    
+def pad_im(image,max_dim,pad_val):
     """
-    im = data.get_data()
-    fig = plot_rgb(im)
-    fname = os.path.split(fname)[1].split(".")[0] + ".png"
-    plt.savefig(outdir + fname, format="png")
-
+    Pads an image to be same dimensions as given max_dim
+    
+    Parameters
+    -----------
+    image: 3-d RGB np array of image slice
+    max_dim: dimension to pad up to
+    pad_val: value to pad with
+    
+    Returns
+    -----------
+    padded_image: 3-d RGB np array of image slice with padding
+    """
+    #pad only in first two dimensions not in rgb
+    pad_width = (((max_dim-image.shape[0])//2,(max_dim-image.shape[0])//2),((max_dim-image.shape[1])//2,(max_dim-image.shape[1])//2),(0,0))
+    padded_image = np.pad(image, pad_width=pad_width, mode='constant', constant_values=pad_val)
+    
+    return padded_image
 
-def plot_rgb(im):
-    plt.rcParams.update({"axes.labelsize": "x-large", "axes.titlesize": "x-large"})
+def create_qa_figure(peak_dirs,peak_values,output_dir,model):
+    """
+    Creates a 9x9 figure of the 3-d volume and saves it
+    
+    Parameters
+    -----------
+    peak_dirs: np array of peak_dirs 
+    peak_values: np array of peak_values
+    output_dir: location to save qa figure
+    model: model type (CSA, CSD)
+    """
+    #set shape of image
+    im_shape = peak_dirs.shape[0:3]
+    im_shape_rgb = im_shape + (3,)
+    max_dim = max(im_shape) 
+
+    #title
+    title = f'QA for Tractography {model.upper()} Model Peak Directions. Brain Volume: {im_shape}'
+    
+    #generate 3-d directional data
+    centers,directions,directions_colors,heights = generate_3_d_directions(peak_dirs,peak_values)
+    
+    #reshape back into a 3-d volume with voxel encoded RGB values corresponding to directional vectors
+    im = directions_colors.reshape(im_shape_rgb)
+    
+    x = [int(im_shape[0] * 0.35), int(im_shape[0] * 0.51), int(im_shape[0] * 0.65)]
+    y = [int(im_shape[1] * 0.35), int(im_shape[1] * 0.51), int(im_shape[1] * 0.65)]
+    z = [int(im_shape[2] * 0.35), int(im_shape[2] * 0.51), int(im_shape[2] * 0.65)]
```

Comment:
how did you pick `0.35`, `0.51`, and `0.65`? arbitrarily, trial and error, ...?

note that it's standard to implement immutable value sets like this as a `tuple` rather than a `list`, (to indicate that you don't plan on changing `x`, `y`, or `z`), so use a tuple here

Reply:
This is from the old qa code, this is how they scaled the shapes.

Reply:
ok, fine then